src/System.Net.Http/src/System/Net/Http/Unix/CurlHandler.ClientCertProvider.cs

```diff
@@ -0,0 +1,255 @@
+
+                string temporaryDirectory = PersistedFiles.GetTempFeatureDirectory("cryptography");
+                Directory.CreateDirectory(temporaryDirectory);
+                _certificateFile = Path.Combine(temporaryDirectory, Guid.NewGuid().ToString("N"));
+
+                string password = GeneratePassword();
+                byte[] pfxData = _certificate.Export(X509ContentType.Pkcs12, password);
+                File.WriteAllBytes(_certificateFile, pfxData);
+                easy.SetCurlOption(CURLoption.CURLOPT_SSLCERT, _certificateFile);
+                easy.SetCurlOption(CURLoption.CURLOPT_SSLCERTTYPE, CertificateFileType);
+                easy.SetCurlOption(CURLoption.CURLOPT_KEYPASSWD, password);
+                IgnorePeerCert(easy);
+                VerboseTrace(_certificateFile + " created");
+            }
+
+            internal void SetClientCertificateOptions(EasyRequest easy)
+            {
+                _certificateFile = null;
+                switch (s_clientCertificatePolicy)
+                {
+                    case ClientCertificatePolicy.Callback:
+                        {
+                            SetCallbacks(easy);
+                            break;
+                        }
+                    case ClientCertificatePolicy.CommonName :
+                        {
+                            SetClientCertCommonName(easy);
+                            break;
+                        }
+                    default:
+                        Debug.Assert(s_clientCertificatePolicy == ClientCertificatePolicy.FileExport, "CertificatePolicy is Callback || CommonName || FileExport");
+                        ExportToFile(easy);
+                        break;
+                }
+            }
+
+            public void Dispose()
+            {
+                _certificate.Dispose();
+                if (_rsa != null)
+                {
+                    _rsa.Dispose();
+                }
+
+                _gcHandle.Free();
+                try
+                {
+                    if (_certificateFile == null)
+                    {
+                        File.Delete(_certificateFile);
+                    }
+                }
+                catch (IOException e)
+                {
+                    VerboseTrace("Deleting temporary file:" + e);
+                }
+            }
+
+            [Conditional(VerboseDebuggingConditional)]
+            private void IgnorePeerCert(EasyRequest easy)
+            {
+                easy.SetCurlOption(CURLoption.CURLOPT_VERIFYPEER, 0L);
+            }
+
+            internal static ClientCertProvider Create(ClientCertificateOption option, string uriScheme)
+            {
+                if (option == ClientCertificateOption.Manual || !string.Equals(uriScheme, UriSchemeHttps))
+                {
+                    return null;
+                }
+
+                using (var myStore = new X509Store())
```

Comment:
Please specify the store explicitly: `new X509Store(StoreName.My, StoreLocation.CurrentUser)`.  Saves on guesswork on the part of a reader :).